File: frontend/lib/core/network/response/bus_calling_point_item_response.dart
class BusCallingPointItemResponse {

  final int sequence;
  final String busStopId;
  final DateTime scheduledTime;
  final double latitude;
  final double longitude;
  final String name;

  BusCallingPointItemResponse._({
    required this.sequence,
    required this.busStopId,
    required this.scheduledTime,
    required this.latitude,
    required this.longitude,
    required this.name,
  });

  factory BusCallingPointItemResponse.fromJson(Map<String, dynamic> json) {
    return BusCallingPointItemResponse._(
      sequence: json['sequence'] as int,
      busStopId: json['busStopId'] as String,
      scheduledTime: DateTime.parse(json['scheduledTime'] as String),
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      name: json['name'] as String,
    );
  }
}

File: Backend/Backend/Models/Response/Bus/BusCallingPointItemResponse.cs
namespace Backend.Models
{
    public record BusCallingPointItemResponse: IResponse
    {
        public required int Sequence { get; init; }

        public required DateTime ScheduledTime { get; init; }

        public required decimal Latitude { get; init; }
        public required decimal Longitude { get; init; }

        public required string Name { get; init; }
    }
}
